### PR TITLE
Fixed #31826 -- Made AlterField operation a noop when adding db_column.

### PR DESCRIPTION
**Pull request by iurisilvio**
https://code.djangoproject.com/ticket/31826

### CODE REVIEW

**Comment by felixxm** [comment]
@iurisilvio Thanks :+1: I was able to reproduce this behavior only on SQLite. IMO we shouldn't add this optimization to the `AlterField` operations. It looks that both issues (ticket-31826, ticket-31825) can be fixed in the SQLite backend, see [comment](https://github.com/django/django/pull/13236#issuecomment-665584461).

**Comment by iurisilvio** [comment]
Same thing about sqlite3 on this PR, it fails with foreign keys on postgres.

I moved my implementation to base schema `alter_field`. I don't have to copy it to sqlite3 schema because their behavior is different only on field name changes.

Also, added tests to the foreign key case.

This PR is ready for another review, but it conflicts with #13236, so I prefer the RenameField PR merged first.

**Comment by iurisilvio** [comment]
I updated this PR with current master and moved the change to the `_field_should_be_altered`.

File: django/db/backends/base/schema.py [hunk 1]
```diff
@@ -1037,11 +1037,28 @@ def _field_indexes_sql(self, model, field):
...
+            changed_kwargs = set(new_kwargs) - set(old_kwargs)
+            # Attrs only in old_field or in both with different values.
+            for old_key, old_value in old_kwargs.items():
+                if old_key not in new_kwargs or old_value != new_kwargs[old_key]:
+                    changed_kwargs.add(old_key)
+            # Only db_column changed and we know field column did not changed.
+            if changed_kwargs == {'db_column'}:
+                return False
+
+        return True
```
Comment (felixxm) [inline django/db/backends/base/schema.py:1061]: We've already compare columns in:
```python
old_field.column != new_field.column or
```
so we can simply ignore `db_column` in kwargs, e.g.
```python
def _field_should_be_altered(self, old_field, new_field):
    _, old_path, old_args, old_kwargs = old_field.deconstruct()
    _, new_path, new_args, new_kwargs = new_field.deconstruct()
    # Ignore db_column to not alter when changing only a db_column but it's
    # the same as a field name.
    old_kwargs.pop('db_column', None)
    new_kwargs.pop('db_column', None)
    # Don't alter when changing only a field name.
    return (
        old_field.column != new_field.column or
        (old_path, old_args, old_kwargs) != (new_path, new_args, new_kwargs)
    )
```

Comment (iurisilvio) [reply django/db/backends/base/schema.py]: Great solution! Thanks! I didn't rethink the past implementation. Updated. :+1:

**Comment by felixxm** [comment]
@iurisilvio Thanks :+1: I pushed small edits.

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Pull request merged**